implement-guess-user-can-add-items-to-cart

scottalexandra opened the pull request:
implemented adding items to cart, added cart to be displayed in menu layout.

scottalexandra committed:
implement-guess-user-can-add-items-to-cart

scottalexandra commented:
Had to modify a few of the tests in cart_spec.rb to align with the way cart needs to be displayed. The _menu partial needs some refactoring, logic can be pushed down to the cart model.

scottalexandra committed:
fixed hound issues, removed dugging stuff

scottalexandra committed:
refactored menu partial to move logic to cart.rb

scottalexandra commented:
refactored _menu partial to remove logic

kpearson commented on app/models/cart.rb:
```diff
@@ -2,11 +2,15 @@ class Cart
 
   def initialize(data)
-    @data = data || Hash.new
+    @data = data || { "" => 0 }
```
Why do we need an default empty string value?

scottalexandra replied:
I wanted to have zero displayed in the cart before items are added. Didn't know the best way to do this...

scottalexandra committed:
refactored add-item-implementation

scottalexandra commented:
Changed cart.new to instantiate a new hash before items are added to cart instead of hard-coding 0. Modified category index view div ids to be name after the category name and fixed tests. Added test for count method, and modified count method to have an initial value of 0.

kpearson commented:
Looks good. Thanks for the revisions.
